### PR TITLE
Fixed the problem of randomly generating garbage in each directory when entering the working directory subdirectory, and cleaning the garbage during rmvenv

### PR DESCRIPTION
**Pull request by Mejituu**
Fixed the problem of randomly generating garbage in each directory when entering the working directory subdirectory, and cleaning the garbage during rmvenv

### CODE REVIEW

**Comment by Mejituu** [comment]
For example AUTOSWITCH_VIRTUAL_ENV_DIR=".virtualenv"

**Comment by Mejituu** [comment]
> For example AUTOSWITCH_VIRTUAL_ENV_DIR=".virtualenv"
Entering any subdirectory of the working directory will generate a .virtualenv in the current directory, even if activate and deactivate are the same.

File: autoswitch_virtualenv.plugin.zsh [hunk 1]
```diff
@@ -264,6 +264,9 @@ function rmvenv()
             # https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch03s04.html
             # https://www.freedesktop.org/wiki/Software/systemd/TheCaseForTheUsrMerge/
             /bin/rm -rf "$(_virtual_env_dir "$venv_name")"
+            if [[ -z "$(/bin/ls -A $VIRTUAL_ENV_DIR)" ]]; then
```
Comment (MichaelAquilina) [inline autoswitch_virtualenv.plugin.zsh:267]: $VIRTUAL_ENV_DIR should probably wrapped in quotes for safety

File: autoswitch_virtualenv.plugin.zsh [hunk 2]
```diff
@@ -23,8 +25,6 @@ function _validated_source() {
 
 function _virtual_env_dir() {
     local venv_name="$1"
-    local VIRTUAL_ENV_DIR="${AUTOSWITCH_VIRTUAL_ENV_DIR:-$HOME/.virtualenvs}"
-    mkdir -p "$VIRTUAL_ENV_DIR"
```
Comment (MichaelAquilina) [inline autoswitch_virtualenv.plugin.zsh:27]: nice catch actually, I didnt even realise this was running here!

**Comment by MichaelAquilina** [comment]
testing this out locally for the day to see if I can catch any unexpected bugs. Wondering if some code has become implicitly reliant on this "bug" as a feature after being here for so long

**Comment by MichaelAquilina** [comment]
looks good. thanks for this!

**Pull request merged**